lisa/nic.py

```diff
@@ -403,56 +379,116 @@ def _get_node_nic_info(self) -> None:
+                        used_module = lspci.get_used_module(pci_slot)
+                        self.append(
+                            NicInfo(
+                                name=nic_name,
+                                lower=lower,
+                                pci_slot=pci_slot,
+                                lower_module_name=used_module,
+                            )
+                        )
                         break
 
         # Collects NIC info for any unpaired NICS
         for nic_name in [
             x
-            for x in self.nic_names
-            if x not in self.get_upper_nics() and x not in self.get_lower_nics()
+            for x in self._nic_names
+            if x not in self.nics.keys() and x not in self.get_lower_nics()
         ]:
-            nic_info = NicInfo(nic_name)
+            nic_info = NicInfo(name=nic_name)
             self.append(nic_info)
 
         assert_that(len(self)).described_as(
             "During Lisa nic info initialization, Nics class could not "
             f"find any nics attached to {self._node.name}."
         ).is_greater_than(0)
 
-    def _get_default_nic(self) -> None:
-        cmd = "/sbin/ip route"
-        ip_route_result = self._node.execute(cmd, shell=True, sudo=True)
-        ip_route_result.assert_exit_code()
-        assert_that(ip_route_result.stdout).is_not_empty()
-        dev_match = self.__dev_regex.search(ip_route_result.stdout)
-        if not dev_match or not dev_match.groups():
-            raise LisaException(
-                "Could not locate default network interface"
-                f" in output:\n{ip_route_result.stdout}"
+        # handle situation when there is no mana driver, but have mana pci devices
+        if self._is_mana_device_discovered() and not self._node.tools[
+            KernelConfig
+        ].is_enabled("CONFIG_MICROSOFT_MANA"):
+            pci_devices = lspci.get_devices_by_type(
+                constants.DEVICE_TYPE_SRIOV, force_run=True
             )
-        assert_that(dev_match.groups()).is_length(1)
-        default_interface_name = dev_match.group(1)
-        assert_that(default_interface_name in self.nic_names).described_as(
+            for pci_device in pci_devices:
+                for nic in self.get_unpaired_devices():
+                    self.nics[nic].pci_slot = pci_device.slot
+                break
+
+    def _is_mana_device_discovered(self) -> bool:
+        lspci = self._node.tools[Lspci]
+        pci_devices = lspci.get_devices_by_type(
+            constants.DEVICE_TYPE_SRIOV, force_run=True
+        )
+        all_mana_devices = False
+        for pci_device in pci_devices:
+            if (
+                "Device 00ba" in pci_device.device_info
+                and pci_device.vendor == "Microsoft Corporation"
+            ):
+                all_mana_devices = True
+            else:
+                all_mana_devices = False
+                break
```

Comment:
It looks should break, when found any `True`, instead of `False`.

Reply:
updated